src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -78,7 +83,12 @@ static TracerProviderSdk()
                 // Callback when Activity is stopped.
                 ActivityStopped = (activity) =>
                 {
-                    if (!Sdk.SuppressInstrumentation && activity.IsAllDataRequested)
+                    if (!activity.IsAllDataRequested)
+                    {
+                        return;
+                    }
+
+                    if (SuppressInstrumentationScope.TryLeave() == 0)

```

Comment:
Should be `if (SuppressInstrumentationScope.TryLeave() <= 0)` no?

Reply:
It is intentional, I've added some comment to clarify the logic.